[TensorExpr] AOT Compiler: support symbolic shape arguments.

ZolotukhinM opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #70475
* #70375
* __->__ #70374



Differential Revision: [D33303646](https://our.internmc.facebook.com/intern/diff/D33303646)

Commit:
[WIP][TensorExpr] AOT Compiler: support symbolic shape arguments.

[ghstack-poisoned]

ZolotukhinM commented:
@ZolotukhinM has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D33303646).

Commit:
Update on "[WIP][TensorExpr] AOT Compiler: support symbolic shape arguments."




Differential Revision: [D33303646](https://our.internmc.facebook.com/intern/diff/D33303646)

[ghstack-poisoned]

ZolotukhinM commented:
@ZolotukhinM has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D33303646).

Commit:
Update on "[WIP][TensorExpr] AOT Compiler: support symbolic shape arguments."




Differential Revision: [D33303646](https://our.internmc.facebook.com/intern/diff/D33303646)

[ghstack-poisoned]

ZolotukhinM commented:
@ZolotukhinM has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D33303646).

Commit:
Update on "[WIP][TensorExpr] AOT Compiler: support symbolic shape arguments."




Differential Revision: [D33303646](https://our.internmc.facebook.com/intern/diff/D33303646)

[ghstack-poisoned]

ZolotukhinM commented:
@ZolotukhinM has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D33303646).

Commit:
Update on "[WIP][TensorExpr] AOT Compiler: support symbolic shape arguments."




Differential Revision: [D33303646](https://our.internmc.facebook.com/intern/diff/D33303646)

[ghstack-poisoned]

ZolotukhinM commented:
@ZolotukhinM has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D33303646).

Commit:
Update on "[TensorExpr] AOT Compiler: support symbolic shape arguments."




Differential Revision: [D33303646](https://our.internmc.facebook.com/intern/diff/D33303646)

[ghstack-poisoned]

ZolotukhinM commented:
@ZolotukhinM has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D33303646).

Commit:
Update on "[TensorExpr] AOT Compiler: support symbolic shape arguments."




Differential Revision: [D33303646](https://our.internmc.facebook.com/intern/diff/D33303646)

[ghstack-poisoned]

ZolotukhinM commented:
@ZolotukhinM has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D33303646).

Commit:
clang-tidy on "[TensorExpr] AOT Compiler: support symbolic shape arguments."




Differential Revision: [D33303646](https://our.internmc.facebook.com/intern/diff/D33303646)

[ghstack-poisoned]

Commit:
clang-tidy on "[TensorExpr] AOT Compiler: support symbolic shape arguments."




Differential Revision: [D33303646](https://our.internmc.facebook.com/intern/diff/D33303646)

[ghstack-poisoned]

ZolotukhinM commented:
@ZolotukhinM has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D33303646).

navahgar commented on torch/csrc/jit/mobile/nnc/context.cpp:
```diff
@@ -224,26 +233,39 @@ c10::impl::GenericList Function::run(
+
+  for (const auto i : c10::irange(sym_shape_positions_.size())) {
+    const auto& sym_shape_pos = sym_shape_positions_[i];
+    const c10::IValue& input_tensor = inputs[sym_shape_pos.input_idx_];
```
nit: s/input_tensor/input (since it is not a tensor)

navahgar commented on torch/csrc/jit/mobile/nnc/context.cpp:
```diff
@@ -224,26 +233,39 @@ c10::impl::GenericList Function::run(
+    const auto& sym_shape_pos = sym_shape_positions_[i];
+    const c10::IValue& input_tensor = inputs[sym_shape_pos.input_idx_];
+    auto dim = input_tensor.toTensor().size(sym_shape_pos.dim_idx_);
+    scalar_values.push_back(dim);
```
Can we add `scalar_values.reserve(sym_shape_positions_.size())` before this loop, since we know the expected size?

navahgar reviewed (approved):
LGTM

Commit:
Update on "[TensorExpr] AOT Compiler: support symbolic shape arguments."




Differential Revision: [D33303646](https://our.internmc.facebook.com/intern/diff/D33303646)

[ghstack-poisoned]

ZolotukhinM commented:
@ZolotukhinM has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D33303646).

Commit:
Update on "[TensorExpr] AOT Compiler: support symbolic shape arguments."




Differential Revision: [D33303646](https://our.internmc.facebook.com/intern/diff/D33303646)

[ghstack-poisoned]

ZolotukhinM commented:
@ZolotukhinM has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D33303646).
